source/slang/slang-check-impl.h

```diff
@@ -899,42 +899,43 @@ namespace Slang
 
             // The generic declaration whose parameters we
             // are trying to solve for.
-            RefPtr<GenericDecl> genericDecl;
+            GenericDecl* genericDecl = nullptr;
 
             // Constraints we have accumulated, which constrain
             // the possible arguments for those parameters.
             List<Constraint> constraints;
         };
 
-        RefPtr<Type> TryJoinVectorAndScalarType(
-            RefPtr<VectorExpressionType> vectorType,
-            RefPtr<BasicExpressionType>  scalarType);
+        Type* TryJoinVectorAndScalarType(
+            VectorExpressionType* vectorType,
+            BasicExpressionType*  scalarType);
 
         struct TypeWitnessBreadcrumb
         {
             TypeWitnessBreadcrumb*  prev;
 
-            RefPtr<Type>            sub;
-            RefPtr<Type>            sup;
+            Type*            sub = nullptr;
+            Type*            sup = nullptr;
             DeclRef<Decl>           declRef;
         };
 
         // Create a subtype witness based on the declared relationship
         // found in a single breadcrumb
-        RefPtr<DeclaredSubtypeWitness> createSimpleSubtypeWitness(
+        DeclaredSubtypeWitness* createSimpleSubtypeWitness(
             TypeWitnessBreadcrumb*  breadcrumb);
 
-            /// Create a withness that `subType` is a sub-type of `superTypeDeclRef`.
-            ///
-            /// The `inBreadcrumbs` parameter represents a linked list of steps
-            /// in the process that validated the sub-type relationship, which
-            /// will be used to inform the construction of the witness.
-            ///
-        RefPtr<Val> createTypeWitness(
-            RefPtr<Type>            subType,
-            DeclRef<AggTypeDecl>    superTypeDeclRef,
-            TypeWitnessBreadcrumb*  inBreadcrumbs);
+        /// Create a withness that `subType` is a sub-type of `superTypeDeclRef`.
+        ///
+        /// The `inBreadcrumbs` parameter represents a linked list of steps
+        /// in the process that validated the sub-type relationship, which
+        /// will be used to inform the construction of the witness.
+        ///
 
```

Comment:
Indenting and spacing changed here from the original.

(Also I now see a really glaring typo in my comment: `withness`).